### PR TITLE
[GH-1682] Add functionality to quests screen

### PR DESCRIPTION
**Pull request by alfopisano**
Issue #1682

## Motivation
Now that the we have the quests screen created we need to add functionality to it: 
- [ ] When completed change to _ready to claim_ version. 
- [ ] When claiming a completed quest the reward should be added to the total of trophies and "a new quest" should be displayed (see disclaimer).
- [ ] When pushing the reroll button the available quests for reroll should be highlighted. 
- [ ] When rerolling a quest "a new quest" should be displayed (see disclaimer). 
- [ ] When pushing the finish button the quests should look like in the beginning. 
- [ ] Show actual amount of quests available for reroll.

DISCLAIMER: since the backend for the quests is not implemented yet, after claiming or rerolling a quest we will display the same as before but as if not started yet. 

### When Quest ready to claim: 
Screenshot 2024-05-07 at 18 27 01

### When in Reroll section:
Screenshot 2024-05-07 at 18 27 10

### After rerolling a quest: 
Screenshot 2024-05-07 at 18 27 18

## Summary of changes
- Created two new scripts "Quest.cs" and "QuestsManager.cs" to add the specified functionality. 

## Checklist
- [x] I have tested the changes locally.
- [x] I have tested the whole game after applying the changes, not only the affected areas.
- [x] I self-reviewed the changes on GitHub, line by line.
- [ ] I have tested the changes in another devices.
  - [ ] Tested in iOS.
  - [ ] Tested in Android.
- [ ] This change requires new documentation.
  - [ ] Documentation has been added/updated.

### CODE REVIEW

**Review by jazimp — CHANGES_REQUESTED** [review]
In the Reroll section, the white glow is missing on the hexagon of the quest that is available for reroll

**Comment by alfopisano** [comment]
> I'm not sure if it's an error due to the quality of the image or the implementation, but I think the progress bars for each quest are distorted (the right tip)

You are right. I think it is due to the stretching for representing the progress. I would do some research on how to avoid this!!

**Comment by jazimp** [comment]
Sorry, I meant to say "After rerolling a quest" instead of "Reroll section". I was referring to the hexagon with the image of H4ck (in this case)
Implementation vs Design: 
Captura de pantalla 2024-05-07 a la(s) 12 34 41
Captura de pantalla 2024-05-07 a la(s) 12 35 06

**Comment by alfopisano** [comment]
I would now make some changes regarding the performance and understanding of the code added.

File: client/Assets/Scripts/UI/Quest.cs [hunk 1]
```diff
@@ -0,0 +1,97 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+using UnityEngine.EventSystems;
+using TMPro;
+
+public class Quest : MonoBehaviour
+{
+    public int reward;
```
Comment (BertovDev) [inline client/Assets/Scripts/UI/Quest.cs:10]: Variables like this where you only access it in the same script can be set as `private` and add the 
`[SerializeField] ` tag to access them from unity. Correct them if you can.

File: client/Assets/Scripts/UI/QuestsManager.cs [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    {
+        int rerollAmount = GetRerollAmount();
+        rerollAvailable.text = rerollAmount + "/6 reroll available";
+    }
+
+    public void ShowReroll() 
+    {
+        rerollActive = true;
+
+        foreach (Quest q in quests)
```
Comment (BertovDev) [inline client/Assets/Scripts/UI/QuestsManager.cs:26]: We could change the `q` to `quest` it is just a readability and standard things. Wdyt?

**Review by BertovDev — CHANGES_REQUESTED** [review]
Really nice and also some minor changes.

- Check the position of the first image in the quest1. It is different from others.

Screenshot 2024-05-13 at 1 08 31 PM

- Notice that the ready to claim container is not clickeable for reclaiming the quest. This is because is a slider and it is set as no interactable. We can do a workaround this or change to a image with a fill type.

File: client/Assets/Scripts/UI/Quest.cs (hunk 1 above)
Comment (alfopisano) [reply client/Assets/Scripts/UI/Quest.cs:10]: Done!!

File: client/Assets/Scripts/UI/QuestsManager.cs (hunk 2 above)
Comment (alfopisano) [reply client/Assets/Scripts/UI/QuestsManager.cs:26]: Done!!

**Review by BertovDev — APPROVED** [review]
Awesome work

**Review by jazimp — APPROVED** [review]

**Review by AngieDutra — APPROVED** [review]
Awesome! 
My only comment is this container may be too small but for now it works.
Checked the responsiveness and is awesome too!
Screenshot 2024-05-13 at 17 38 54

**Pull request merged**